packages/portfolio-deploy/test/portfolio-start.core.test.ts

```diff
@@ -216,52 +247,80 @@ test('delegate ymax control', async t => {
-          },
-        });
-      },
-    }),
-  );
+  const addrCtrl = 'agoric1ymaxcontrol';
+  const [walletCtrl] = await provisionSmartWallet(addrCtrl);
 
   const zone = rootZone.subZone('bootstrap vat');
-  powers.produce.ymax0Kit.resolve({
-    creatorFacet: Far('mockCreator'),
-    adminFacet: Far('mockAdmin'),
-    instance: Far('mockInstance'),
-  } as any);
   await delegatePortfolioContract(
     // @ts-expect-error mock
     { ...powers, zone },
-    { options: { ymaxControlAddress: addr } },
+    { options: { ymaxControlAddress: addrCtrl } },
   );
-  const ctrl = await ctrlKit.promise;
+  await eventLoopIteration(); // core eval doesn't block on delivery
+
+  t.log('redeem ymaxControl invitation');
+  await E(E(walletCtrl).getOffersFacet()).executeOffer({
+    id: 0,
+    invitationSpec: {
+      source: 'purse',
+      description: 'deliver ymaxControl',
+      instance: pInst,
+    },
+    proposal: {},
+    saveResult: { name: 'ymaxControl' },
+  });
+
+  t.log('getDepositFacet');
+  await E(E(walletCtrl).getInvokeFacet()).invokeEntry({
+    targetName: 'ymaxControl',
+    method: 'getCreatorFacet',
+    args: [],
+    saveResult: { name: 'ymax0.creatorFacet' },
+  });
+
+  t.log('invite planner');
+  const addrPl = 'agoric1planner';
+  const [walletPl] = await provisionSmartWallet(addrPl);
+  await E(E(walletCtrl).getInvokeFacet()).invokeEntry({
+    targetName: 'ymax0.creatorFacet',
+    method: 'deliverPlannerInvitation',
+    args: [addrPl, pInst],
+  });
+  await eventLoopIteration(); // wait for vow to settle
+
+  t.log('redeem planner invitation');
+  const yInst = await E(agoricNames).lookup('instance', contractName);
+  await E(E(walletPl).getOffersFacet()).executeOffer({
+    id: 0,
+    invitationSpec: {
+      source: 'purse',
+      description: 'planner',
+      instance: yInst,
+    },
+    proposal: {},
+    saveResult: { name: 'planner' },
+  });
 
-  // eslint-disable-next-line no-underscore-dangle
-  t.deepEqual(await E(ctrl as any).__getMethodNames__(), [
-    '__getInterfaceGuard__',
-    '__getMethodNames__',
-    'getCreatorFacet',
-    'getPublicFacet',
-    'install',
-    'installAndStart',
-    'pruneChainStorage',
-    'start',
-    'terminate',
-    'upgrade',
-  ]);
+  t.log('make a portfolio');
+  const { poc26 } = common.brands;
+  const accessToken = poc26.mint.mintPayment(poc26.make(1n));
+  const [walletTrader] = await provisionSmartWallet('agoric1trader1');
+  await E(E(walletTrader).getDepositFacet()).receive(accessToken);
+  await E(E(walletTrader).getOffersFacet()).executeOffer({
+    id: 1,
+    invitationSpec: {
+      source: 'contract',
+      instance: yInst,
+      publicInvitationMaker: 'makeOpenPortfolioInvitation',
+    },
+    proposal: { give: { Access: poc26.make(1n) } },
+    offerArgs: {},
+  });
+
+  t.log('invoke planner');
+  await E(E(walletPl).getInvokeFacet()).invokeEntry({
+    targetName: 'planner',
+    method: 'submit',
+    args: [0, []],
+  });
```

Comment:
@amessbee  the planner facet is somewhat analagous to the resolver facet; this one takes advantage of the upgraded smartWallet to avoid the `invitationMakers` dance.